src/core/helpers.rs

```diff
@@ -94,6 +94,49 @@ pub fn keycodes_from_xmodmap() -> CodeMap {
     }
 }
 
+/**
+ * Run the xmodmap command to dump the system modifier key table
+ */
+pub fn modifiers_from_xmodmap() -> ModMap {
+    let mut mod_map = ModMap::new();
+
+    match Command::new("xmodmap").arg("-pm").output() {
+        Err(e) => panic!("unable to fetch modifiers via xmodmap: {}", e),
+        Ok(o) => match String::from_utf8(o.stdout) {
+            Err(e) => panic!("invalid utf8 from xmodmap: {}", e),
+            Ok(s) => s.lines().skip(2).for_each(|l| {
+                // words == <mod name> <keyname (keycode), ...>
+                let mut words = l.split_whitespace();
+                let mask = words.next().map(|name| match name {
+                    "shift" => xcb::MOD_MASK_SHIFT,
+                    "lock" => xcb::MOD_MASK_LOCK,
+                    "control" => xcb::MOD_MASK_CONTROL,
+                    "mod1" => xcb::MOD_MASK_1,
+                    "mod2" => xcb::MOD_MASK_2,
+                    "mod3" => xcb::MOD_MASK_3,
+                    "mod4" => xcb::MOD_MASK_4,
+                    "mod5" => xcb::MOD_MASK_5,
+                    _ => panic!("unknown modifier: {}", name),
```

Comment:
we probably don't want to panic on an unknown modifier. It should be safe to just drop it, probably with a `warn` level log of the name? That way, if we do drop something that later on someone wants to use, they are able to tell us exactly what it is from the log.